backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py

```diff
@@ -0,0 +1,192 @@
+
+    def _get_revert_pr_mapping_for_original_prs(
+        self, prs: List[PullRequest]
+    ) -> List[PullRequestRevertPRMapping]:
+        """
+        This function takes a list of PRs and for each PR it tries to
+        find if that pr has been reverted and by which PR. It is done
+        by taking repo_id and the merge commit hash and searching for
+        the string 'revert-[merge-commit-hash]' in the head branch.
+        """
+
+        repo_ids: Set[str] = set()
+        repo_id_to_pr_merge_hash_to_id_map: Dict[str, Dict[str, str]] = {}
+        pr_merge_hash_match_strings: List[str] = []
+
+        for pr in prs:
+            if pr.state != PullRequestState.MERGED:
+                continue
+
+            pr_merge_hash_match_strings.append(f"revert-{pr.merge_commit_sha[:8]}")
+            repo_ids.add(str(pr.repo_id))
+
+            if str(pr.repo_id) not in repo_id_to_pr_merge_hash_to_id_map:
+                repo_id_to_pr_merge_hash_to_id_map[str(pr.repo_id)] = {}
+
+            repo_id_to_pr_merge_hash_to_id_map[str(pr.repo_id)][
+                str(pr.merge_commit_sha[:8])
+            ] = pr.id
+
+        if len(pr_merge_hash_match_strings) == 0:
+            return []
+
+        revert_prs: List[PullRequest] = (
+            self.code_repo_service.get_prs_by_head_branch_match_strings(
+                list(repo_ids), pr_merge_hash_match_strings
+            )
+        )
+
+        revert_pr_mappings: List[PullRequestRevertPRMapping] = []
+
+        for rev_pr in revert_prs:
+            merge_commit_hash = self.get_revert_merge_commit_hash(rev_pr.head_branch)
+            if merge_commit_hash is None:
+                continue
+
+            repo_key_exists = repo_id_to_pr_merge_hash_to_id_map.get(
+                str(rev_pr.repo_id)
+            )
+            if repo_key_exists is None:
+                continue
+
+            original_pr_id = repo_id_to_pr_merge_hash_to_id_map[
+                str(rev_pr.repo_id)
+            ].get(merge_commit_hash)
+            if original_pr_id is None:
+                continue
+
+            revert_pr_mp = PullRequestRevertPRMapping(
+                pr_id=rev_pr.id,
+                actor_type=PullRequestRevertPRMappingActorType.SYSTEM,
+                actor=None,
+                reverted_pr=original_pr_id,
+                updated_at=datetime.now(),
+            )
+            revert_pr_mappings.append(revert_pr_mp)
+
+        return revert_pr_mappings
+
+    def _get_revert_pr_mapping_for_revert_prs(
+        self, prs: List[PullRequest]
+    ) -> List[PullRequestRevertPRMapping]:
+        """
+        This function takes a list of PRs and for each PR it tries to
+        find if that pr is a revert PR and for that revert PR it tries
+        to find the original which was reverted and create a mapping.
+        """
+        revert_pr_hashes: List[str] = []
+        repo_ids: Set[str] = set()
+        repo_id_to_pr_merge_hash_to_id_map: Dict[str, Dict[str, str]] = {}
```

Comment:
repo_id_to_pr_merge_hash_to_revert_pr_id_map would make this more readable as this code is already kind of complex

Reply:
Yeah, makes sense. Made the changes. Thanks for the suggestion.